### DISCUSSION TITLE
Correct way to set background color/ using skybox causes black flash momentarily

### DISCUSSION THREAD

**Discussion by tomriddle25** [sceneview/sceneview#494] [Q&A]
Hi folks, first of all thanks a lot for building and maintaining `SceneView`, its been a boon for 3d model viewing on a jetpack compose project. 

My problem is that the models I am using do not have a background, and are displayed with a black background, and when I am trying to use the existing solutions discussed on issues like a colored `skybox` , I am getting a black background initially when the composable is rendered and then it becomes white.
Its sort of like a black flash.
```
Scene(
    modifier = Modifier
        .fillMaxSize()
        .background(Color.WHITE),
    engine = engine,
    modelLoader = modelLoader,
    cameraNode = cameraNode,
    childNodes =
    listOf(centerNode,
        rememberNode {
            ModelNode(
                modelInstance = modelLoader.createModelInstance("avatar/compressed_new/St2.glb"),
            )
        }),
    onViewCreated = {
        skybox = Skybox.Builder().color(255f, 255f, 255f, 1.0f).build(this.engine)
    }
)
```

Here's a video of the issue. 

https://github.com/SceneView/sceneview-android/assets/17904436/e8f8b837-9bb6-488d-ae00-99995403af4e

P.S. - I am using the latest release 2.1.0.

**Comment by ThomasGorisse** [comment]
You're not so far from being right but you are applying the `skybox` once the view is created and so the default black `skybox` has already been rendered.
You should instead use the `Scene`'s `environment` param. I know it's not clear but it was originaly made like that to also handle `.ktx` and `.hdr` skyboxes which are also generating an `IndirectLight`.

Solution (I hope):
```kotlin
Scene(
    modifier = Modifier
        .fillMaxSize(),
    engine = engine,
    modelLoader = modelLoader,
    cameraNode = cameraNode,
    childNodes = listOf(
        centerNode,
        rememberNode {
            ModelNode(
                modelInstance = modelLoader.createModelInstance("avatar/compressed_new/St2.glb"),
            )
        }
    ),
    environment = rememberEnvironment(engine ) {
        Environment(
            skybox = Skybox.Builder()
                .color(255f, 255f, 255f, 1.0f)
                .build(engine)
        )
    }
)
```

**Reply by tomriddle25** [reply]
Hey first of all thanks for responding. I missed this. 
But I believe I am using something similar already 

Checkout my environment 

```
val environment = rememberEnvironment(
    engine = engine, 
    environment = SceneView.createEnvironment(
                    engine = engine,
                    isOpaque = true,
                    indirectLight = KTX1Loader.createIndirectLight(
                        environmentLoader.engine,
                        context.assets.readBuffer(
                            fileLocation = "environments/neutral/neutral_ibl.ktx"
                        ),
                    ),
                    skybox = Skybox.Builder().color(255f, 255f, 255f, 1.0f).build(engine),
                )
)
```

But I am still getting the same issue.
Am I missing something?

**Comment by ThomasGorisse** [comment]
The black flash is caused by the skybox loading asynchronously while the scene renders with no background. In v4.0.0-rc.1, you can set an initial solid color:

```kotlin
SceneView(
    environment = rememberEnvironment(
        skyboxColor = Color.White // immediate solid color, no flash
    )
) {
    // your nodes
}
```

For a transparent background, use `isTransparent = true` on SceneView. The solid color appears immediately while any HDR loads in the background, eliminating the flash.
